build-release.sh

```diff
@@ -10,7 +10,8 @@ export OUT_DIR="${1-dist}"
 export VERSION=${2:-$(git describe --tags --always --dirty)}
 
 # To overwrite the version details, pass something as the third arg. Empty string disables it.
-export VERSION_DETAILS=${3-"$(date -u +"%FT%T%z")/$(git describe --tags --always --long --dirty)"}
+export COMMIT_DATE=$(git log -1 --format=%at $VERSION | { read d ; date -u +"%FT%T%z" -d @"$d" ; })
```

Comment:
To be honest ... I am kind of for dropping this :shrug: 

We have the commit - that is all we need. Or we have a really well tagged build - so again the date of the build doesn't really matter.

To be honest I really was hoping that we can get rid of this all together and just rely on ReadBuildInfo, unfortunately `go build` in practice never populates this (see [here](https://github.com/golang/go/issues/51279#issuecomment-1048196896) for reasons). 

So in practice that is not ... possible unless we figure out how to make `go install` compile. Which works *except* that:
1. `-o` does not [work](https://github.com/golang/go/issues/44469#issuecomment-1656803740) :( 
2. `GOBIN` setting does [not work](https://github.com/golang/go/issues/57485) with `GOOS` and friends 


So ... in theory we can have `go install ...; mv $GOBIN/k6 ... somehwere` 

But this seems ... bad - I currently hope that the 57485 will be fixed :crossed_fingers: or `go build` will be allowed to buildinfo.

Currently if you do `go install` vs what we have as binary the output is different:

```
[mstoykov@fedora k6]$ go install go.k6.io/k6@v0.46.0
[mstoykov@fedora k6]$ k6 version
k6 v0.46.0 (v0.46.0, go1.21.1, linux/amd64)
[mstoykov@fedora k6]$ k6.v0.46.0 version
k6 v0.46.0 (2023-08-14T13:23:26+0000/v0.46.0-0-gcbd9e9ad, go1.20.7, linux/amd64)
```

And to be honest I find *our* build to be worse :shrug: 

so .. to reiterate ... can we drop the date at least?